password-hash/src/lib.rs

```diff
@@ -185,13 +212,14 @@ pub trait McfHasher {
 /// PHC strings have the following format:
 ///
 /// ```text
-/// $<id>[$v=<version>][$<param>=<value>(,<param>=<value>)*][$<salt>[$<hash>]]
+/// $<id>[$v=<version>][$pepper=<pepper_algorithm>][$<param>=<value>(,<param>=<value>)*][$<salt>[$<hash>]]
```

Comment:
How should we call the field? A shorter name would be good but it shouldn't interfere with a `param` key.

Reply:
These fields are part of the [PHC string format](https://github.com/P-H-C/phc-string-format/blob/master/phc-sf-spec.md), so it doesn't make sense to add additional fields to it.

If we are going to include information about pepper, it should probably be part of the algorithm identifier field.

Note that certain algorithms, e.g. Argon2, already have built-in support for pepper.

Reply:
I thought you [pleaded](https://github.com/RustCrypto/traits/issues/694#issuecomment-889138538) for not using the algorithm identifier field as it would violate the PHC format.

What do you recommend?

Reply:
Adding a new bespoke field is worse. Regardless of what the solution is, it shouldn't work outside the scope of the PHC format.

The spec does not encode the pepper into the hash whatsoever.

Reply:
You mean only the hash and no other fields, or?
So it would be pending on a potential PHC update.

Reply:
The spec already mentions pepper. The example Argon2 hash includes pepper, however it does not change how the hash is serialized.